interactive_engine/executor/ffi/src/store/jna_response.rs

```diff
@@ -54,13 +57,26 @@ impl JnaResponse {
         ::std::mem::forget(msg);
     }
 
-    pub fn data(&mut self, data: Vec<u8>) -> GraphResult<()> {
+    pub fn byte_data(&mut self, data: Vec<u8>) -> GraphResult<()> {
         if data.len() != data.capacity() {
             let msg = format!("data len {} must eq capacity {}", data.len(), data.capacity());
             return Err(GraphError::new(GraphErrorCode::InvalidData, msg));
         }
         self.data = data.as_ptr() as *const c_void;
         self.len = data.len() as i32;
+        self.byteLen = data.len() as i32;
+        ::std::mem::forget(data);
+        Ok(())
+    }
+
+    pub fn i32_data(&mut self, data: Vec<i32>) -> GraphResult<()> {
+        if data.len() != data.capacity() {
+            let msg = format!("data len {} must eq capacity {}", data.len(), data.capacity());
+            return Err(GraphError::new(GraphErrorCode::InvalidData, msg));
+        }
+        self.data = data.as_ptr() as *const c_void;
+        self.len = data.len() as i32;
+        self.byteLen = (data.len() * mem::size_of::<i32>()) as i32;
```

Comment:
Can we do the type conversion in the java part? So that we don't need to add a JNA interface every time we want to return a new data type.

Reply:
right